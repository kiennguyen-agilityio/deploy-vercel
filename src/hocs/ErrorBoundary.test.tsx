import { render } from '@testing-library/react';
import ErrorBoundary from './ErrorBoundary';
import { ReactNode } from 'react';

const defaultProps = {
  fallback: <div>Error fallback</div>,
};

const setup = (children: ReactNode, props = {}) => {
  const mergedProps = { ...defaultProps, ...props };
  return render(<ErrorBoundary {...mergedProps}>{children}</ErrorBoundary>);
};

describe('ErrorBoundary component', () => {
  test('renders children when there is no error', () => {
    const { container } = setup(<div>Child component</div>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders fallback when there is an error', () => {
    console.error = jest.fn();

    const ChildComponent = () => {
      throw new Error('Test error');
    };

    const { container } = setup(<ChildComponent />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
