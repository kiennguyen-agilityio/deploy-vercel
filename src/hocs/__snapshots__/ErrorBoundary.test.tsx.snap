// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component renders children when there is no error 1`] = `
<div>
  Child component
</div>
`;

exports[`ErrorBoundary component renders fallback when there is an error 1`] = `
<div>
  Error fallback
</div>
`;
